File: 2018/day-02/code.py
import fileinput
from collections import Counter

lines = list(fileinput.input())

def part1():
    has2 = 0
    has3 = 0
    for line in lines:
        c = Counter(line).values()
        has2 += 2 in c
        has3 += 3 in c
    return has2 * has3

def part2():
    for a in lines:
        for b in lines:
            x = ''.join(a for a,b in zip(a,b) if a == b)
            if len(x) == len(a) - 1:
                return x

print(part1())
print(part2())

File: 2022/day-10/code.js
const utils = require('../../utils');
const INPUT = utils.getInput();

function part1() {
    function tick() {
        cycle++;
        if ((cycle + 20) % 40 === 0) {
            sum += cycle * x;
        }
    }

    let sum = 0
    let x = 1
    let cycle = 0
    for (const command of INPUT) {
        const [ins, value] = command.split(' ')
        if (ins === 'noop') {
            tick();
        } else {
            tick();
            tick();
            x += +value;
        }
    }

    console.log(sum)
}

function part2() {
    function tick() {
        cycle++;
        const ptr = (cycle - 1) % 40;
        display[~~((cycle - 1) / 40)][(cycle - 1) % 40] =
            ptr >= x - 1 && ptr <= x + 1 ? '#' : '.';
    }

    const display = [...Array(6)].map(() => [...Array(40)].fill('.'));
    let x = 1
    let cycle = 0
    for (const command of INPUT) {
        const [ins, value] = command.split(' ')
        if (ins === 'noop') {
            tick();
        } else {
            tick();
            tick();
            x += +value;
        }
    }

    console.log(display.map((line) => line.join('')).join('\n'));
}

part1()
part2()